src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/UploadEnvelopeDocumentsService.java

```diff
@@ -0,0 +1,243 @@
+        String zipFileName,
+        UUID envelopeId
+    ) {
+        try {
+            return Optional.of(blobContainer.getBlockBlobReference(zipFileName));
+        } catch (URISyntaxException | StorageException exception) {
+            log.error(
+                "Unable to get blob client. Container: {}, Blob: {}, Envelope ID: {}",
+                blobContainer.getName(),
+                zipFileName,
+                envelopeId,
+                exception
+            );
+
+            return Optional.empty();
+        }
+    }
+
+    private Optional<BlobInputStream> getBlobInputStream(
+        CloudBlockBlob blobClient,
+        String containerName,
+        String zipFileName,
+        UUID envelopeId
+    ) {
+        try {
+            return Optional.of(blobClient.openInputStream());
+        } catch (StorageException exception) {
+            log.error(
+                "Unable to get blob input stream. Container: {}, Blob: {}, Envelope ID: {}",
+                containerName,
+                zipFileName,
+                envelopeId,
+                exception
+            );
+
+            return Optional.empty();
+        }
+    }
+
+    private Optional<ZipFileProcessingResult> processInputStream(
+        BlobInputStream blobInputStream,
+        String containerName,
+        String zipFileName,
+        UUID envelopeId
+    ) {
+        try (ZipInputStream zis = new ZipInputStream(blobInputStream)) {
+            return Optional.of(
+                zipFileProcessor.process(zis, containerName, zipFileName)
+            );
+        } catch (Exception exception) {
+            log.error(
+                "Failed to process zip. File: {}, Container: {}, Envelope ID: {}",
+                zipFileName,
+                containerName,
+                envelopeId,
+                exception
+            );
+
+            createDocUploadFailureEvent(containerName, zipFileName, exception.getMessage(), envelopeId);
+        }
+
+        return Optional.empty();
+    }
+
+    private boolean uploadParsedZipFileName(Envelope envelope, List<Pdf> pdfs) {
+        try {
+            documentProcessor.uploadPdfFiles(pdfs, envelope.getScannableItems());
+
+            log.info(
+                "Uploaded pdfs. File {}, Container: {}, Envelope ID: {}",
+                envelope.getZipFileName(),
+                envelope.getContainer(),
+                envelope.getId()
+            );
+
+            return true;
+        } catch (Exception exception) {
+            log.error(
+                "Failed to upload PDF files to Document Management. File: {}, Container: {}, Envelope ID: {}",
+                envelope.getZipFileName(),
+                envelope.getContainer(),
+                envelope.getId(),
+                exception
+            );
+
+            envelope.setUploadFailureCount(envelope.getUploadFailureCount() + 1);
+            Status.fromEvent(DOC_UPLOAD_FAILURE).ifPresent(envelope::setStatus);
```

Comment:
what if it's not present? would that be normal?
How about
```java
envelope.setStatus(UPLOAD_FAILURE);
```
>

Reply:
I agree, nothing better than explicit setting. We know what status to set, no need to retrieve it.

Reply:
copy paste methodology used. makes more sense to set directly ;)